File: se2webdriver/src/test/java/com/deepsouthsoftware/uatshootout/TestFindKioskFlow.java
package com.deepsouthsoftware.uatshootout;

import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

public class TestFindKioskFlow {	
	private WebDriver driver;
			
	@Before
	public void setUp() throws Exception {
		driver = new FirefoxDriver();
		driver.get("http://localhost:8080/fluffbox-rwx");
	}

	@Test
	public void testFindKioskFlow() throws Exception {
		driver.findElement(By.xpath("//a[contains(@href, '/fluffbox-rwx/kiosk/find')]"))
					.click();
		driver.findElement(By.id("searchCriteria"))
					.sendKeys("Ft. Lauderdale");
		driver.findElement(By.id("searchButton")).click();
		
		WebElement findSpeakersHereLink = null;
		
		for (int second = 0;; second++) {
			if (second >= 60) 
				fail("timeout");
			try { 
				 findSpeakersHereLink = driver.findElement(By.linkText("Find Speakers Here"));
				 break; 
			} catch (Exception e) {}
			Thread.sleep(1000);
		}
		
		findSpeakersHereLink.click();
		
		driver.findElement(By.linkText("Aleksandar Seovic")).click();
		driver.findElement(By.linkText("RENT NOW!")).click();
		driver.findElement(By.linkText("Continue")).click();
		
		driver.findElement(By.id("username")).sendKeys("joeuser");
		driver.findElement(By.id("password")).sendKeys("password");
		driver.findElement(By.id("loginButton")).click();
		
		assertTrue(driver.getPageSource().indexOf("This concludes your fake rental experience!") != -1);
	}

	@After
	public void tearDown() throws Exception {
		driver.quit();
	}
}

File: watir/test/testfindkioskflow.rb
require 'test/unit'
require 'rubygems'
require 'watir'

class TestFindKioskFlow < Test::Unit::TestCase
  
def setup
  Watir::Browser.default = 'firefox'
  @browser = Watir::Browser.new
  @browser.goto "http://localhost:8080/fluffbox-rwx"
end
  
def test_find_kiosk_flow
  @browser.link(:xpath, "//a[contains(@href, '/fluffbox-rwx/kiosk/find')]").click
  
  @browser.text_field(:id, "searchCriteria").set("Ft. Lauderdale")
  @browser.button(:id, "searchButton").click
  
  until @browser.link(:text, "Find Speakers Here").exists? do
    sleep 0.5
  end
  
  @browser.link(:text, "Find Speakers Here").click
  @browser.link(:text, "Aleksandar Seovic").click
  @browser.link(:text, "RENT NOW!").click
  @browser.link(:text, "Continue").click
  
  assert(@browser.text.include? "This concludes your fake rental experience!")  
end  

def teardown
  @browser.close
end
  
end